src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectNative.cs

```diff
@@ -0,0 +1,363 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace System.Data.SqlClient
+{
+    internal class TdsParserStateObjectNative : TdsParserStateObject
+    {
+        private SNIHandle _sessionHandle = null;              // the SNI handle we're to work on
+
+        private SNIPacket _sniPacket = null;                // Will have to re-vamp this for MARS
```

Comment:
The comment with this field declaration. Looks like a sort of TODO for MARS. 
This has propagated since the time the SqlClient code was ported to CoreFx and has been in the framework code as well. 

I don't think this is going to be changed or revamped.